shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1757,6 +1758,30 @@ bool WebContents::IsCrashed() const {
   return web_contents()->IsCrashed();
 }
 
+void WebContents::CrashProcess() {
+  content::RenderWidgetHostView* view =
+      web_contents()->GetRenderWidgetHostView();
+  if (!view)
+    return;
+
+  content::RenderWidgetHost* rwh = view->GetRenderWidgetHost();
+  if (!rwh)
+    return;
+
+  content::RenderProcessHost* rph = rwh->GetProcess();
+  if (rph) {
+#if defined(OS_LINUX) || defined(OS_CHROMEOS)
+    // A generic |CrashDumpHungChildProcess()| is not implemented for Linux.
+    // Instead we send an explicit IPC to crash on the renderer's IO thread.
+    rph->ForceCrash();
+#else
+    // Try to generate a crash report for the hung process.
+    CrashDumpHungChildProcess(rph->GetProcess().Handle());
+    rph->Shutdown(content::RESULT_CODE_HUNG);
+#endif
```

Comment:
This implementation is consistent with what's in hung_renderer_view.cc and smells OK in terms of working.

I am wondering though what happens to these crash reports? If these get stored on the user's machine, does it make sense to have API that generates a lot of data files when working as intended?

These crash reports could be a boon for well-maintained apps that have people who mine these reports, but would not be great for less-supported apps iff the crash reports just keep piling up on the user's machine.

Reply:
@ckerr for apps that haven't configured crashReporter, no dumps will be generated when crashes occur. https://source.chromium.org/chromium/chromium/src/+/master:components/crash/core/app/crashpad_mac.mm;l=80-81;drc=bc576dbe56e7f97ceb1973c6cd4953afca86f175 the database is only present if `crashReporter.start()` has been called.